performance/server/src/test/java/org/hawkular/apm/performance/server/ServiceTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.performance.server;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.HashMap;
+
+import org.hawkular.apm.api.services.ServiceResolver;
+import org.hawkular.apm.api.services.TracePublisher;
+import org.junit.Test;
+
+/**
+ * @author gbrown
+ */
+public class ServiceTest {
+
+    /**  */
+    private static final String TXN_NAME = "TestBTxnName";
+
+    @Test
+    public void testBusinessTxnNamePropagated() {
+        TestTracePublisher publisher = (TestTracePublisher) ServiceResolver.getSingletonService(TracePublisher.class);
```

Comment:
Understand, but in this case the TracePublisher impl is hidden in the collector mechanism, so not accessible - so by using a known 'test' implementation I'm able to inspect what Traces have been created.